common/analysis/file_analyzer.h

```diff
@@ -87,7 +87,12 @@ std::ostream& operator<<(std::ostream&, const RejectedToken&);
 // FileAnalyzer holds the results of lexing and parsing.
 class FileAnalyzer {
  public:
-  explicit FileAnalyzer(absl::string_view contents, absl::string_view filename)
+  FileAnalyzer(std::shared_ptr<MemBlock> contents, absl::string_view filename)
+      : text_structure_(new TextStructure(std::move(contents))),
```

Comment:
Is the `move` necessary for `shared_ptr`?

Reply:
clang-tidy will otherwise complain on performance grounds. Because without move, we increase the reference counter when copying, and then decrease it again once the `contents` parameter falls out of scope. With `std::move` we just take the already increased reference counter we get from the per-value parameter.